### PR TITLE
Smart Wallet: Custom Error Cleanup

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

Had a chat w/ Claude about how we should cleanup and format our SDK errors.

## Test plan

TS Compiling and CI passing.

### CODE REVIEW

File: packages/client/wallets/smart-wallet/src/types/Error.ts [hunk 1]
```diff
@@ -1,57 +1,51 @@
...
+    UNCATEGORIZED: "smart-wallet:uncategorized", // catch-all error code
+} as const;
+export type SmartWalletErrorCode = (typeof SmartWalletErrors)[keyof typeof SmartWalletErrors];
+
+export class SmartWalletSDKError extends Error {
+    public readonly code: SmartWalletErrorCode;
+
+    constructor(message: string, code: SmartWalletErrorCode = SmartWalletErrors.UNCATEGORIZED) {
         super(message);
-        Object.setPrototypeOf(this, NotAuthorizedError.prototype);
```
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/src/types/Error.ts:6]: We're targeting ES6 and don't need this anymore.

File: packages/client/wallets/smart-wallet/src/types/Error.ts [hunk 2]
```diff
@@ -1,57 +1,51 @@
-export class NotAuthorizedError extends Error {
-    code = "ERROR_NOT_AUTHORIZED";
-
-    constructor(message: string) {
+export const SmartWalletErrors = {
```
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/src/types/Error.ts:1]: To be consistent with our other SDKs

File: packages/client/wallets/smart-wallet/src/blockchain/wallets/service.ts [hunk 3]
```diff
@@ -62,7 +62,9 @@ export class SmartWalletService {
             return new EVMSmartWallet(this.crossmintWalletService, account, publicClient, chain);
         } catch (error: any) {
             if (error.code == null) {
-                throw new WalletSdkError(`Error creating the Wallet ${error?.message ? `: ${error.message}` : ""}`);
+                throw new SmartWalletSDKError(
+                    `Error creating the Wallet ${error?.message ? `: ${error.message}` : ""}`
```
Comment (alfonso-paella) [inline packages/client/wallets/smart-wallet/src/blockchain/wallets/service.ts:66]: I would do:

`Wallet creation failed ${error.message ? `: ${error.message}` : ""}.`

File: packages/client/wallets/smart-wallet/src/types/Error.ts (hunk 2 above)
Comment (alfonso-paella) [reply packages/client/wallets/smart-wallet/src/types/Error.ts:1]: Did we need a not-authorized thats just for smart wallets? Or is there a chance we can have a core error for that across all routes

File: packages/client/wallets/smart-wallet/src/types/Error.ts [hunk 4]
```diff
@@ -1,57 +1,51 @@
...
+    NOT_AUTHORIZED: "smart-wallet:not-authorized",
+    TRANSFER: "smart-wallet:transfer.error",
+    TRANSACTION: "smart-wallet:transaction.error",
+    CROSSMINT_SERVICE: "smart-wallet:crossmint-service.error",
```
Comment (alfonso-paella) [inline packages/client/wallets/smart-wallet/src/types/Error.ts:5]: Same for this

File: packages/client/wallets/smart-wallet/src/types/Error.ts [hunk 5]
```diff
@@ -1,57 +1,51 @@
...
-/**
- * Generic undefined error
- */
-export class WalletSdkError extends Error {
-    code = "ERROR_UNDEFINED";
-
-    constructor(message: string) {
-        super(message);
-        Object.setPrototypeOf(this, WalletSdkError.prototype);
+        super("Smart Wallet SDK should only be used client side.", SmartWalletErrors.RUNNING_ON_SERVER);
```
Comment (alfonso-paella) [inline packages/client/wallets/smart-wallet/src/types/Error.ts:49]: I'm not sure we need an error code for this scenario. To em error codes are useful for errors you must handle (e.g. not authorized maybe, or network errors or whatever), but this one is not an error you must handle - its one you'll get during development once 

So I think the error message is sufficient

**Pull request merged**